src/fault.h

```diff
@@ -0,0 +1,71 @@
+#ifndef FAULT_H
+#define FAULT_H
+
+#include "string_id.h"
+#include "json.h"
+
+class fault;
+using fault_id = string_id<fault>;
+
+class Skill;
+using skill_id = string_id<Skill>;
+
+using quality_id = std::string;
+
+using itype_id = std::string;
+
+class fault
```

Comment:
So you could specify fault repairs using the same syntax as crafting requirements?